pkg/overlay/cache.go

```diff
@@ -243,35 +270,19 @@ func (cache *Cache) FindStorageNodesWithPreferences(ctx context.Context, req Fin
 		UptimeCount:        preferences.UptimeCount,
 		UptimeSuccessRatio: preferences.UptimeRatio,
 
-		Excluded: req.ExcludedNodes,
-
-		MinimumVersion: preferences.MinimumVersion,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	newNodeCount := int64(float64(reputableNodeCount) * preferences.NewNodePercentage)
-	newNodes, err := cache.db.SelectNewStorageNodes(ctx, int(newNodeCount), &NewNodeCriteria{
-		FreeBandwidth: req.FreeBandwidth,
-		FreeDisk:      req.FreeDisk,
-
-		AuditThreshold: preferences.NewNodeAuditThreshold,
-
-		Excluded: req.ExcludedNodes,
+		Excluded: excluded,
 
 		MinimumVersion: preferences.MinimumVersion,
 	})
 	if err != nil {
```

Comment:
I'm wondering whether we should fall back to selecting any node when there aren't enough nodes at this point?

Reply:
No. The callers of this method already expect to get the exact number of the requested number of nodes. No less, no more.

Reply:
I was thinking of when there aren't enough storage nodes that are reputable.

Reply:
Do you mean to complete the number with new vetted nodes?

I am not sure if this makes sense. We should manage our vetting process in a way that we have at least 90-95 reputable nodes in the network.

It is worth discussing this, but I think is out of scope for this PR.

Reply:
Sure let's bring the discussion to design-drafts.